### PR TITLE
BAVL-115 : Get list of courts and probation teams

### PR DESCRIPTION
**Pull request by timharrison-moj**
New controllers and endpoints for obtaining a list of enable courts and probation teams.
Entities, models and mapping functions.
Tests and integration tests.
Also updated the appointment type values in data and enums.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CourtsService.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.CourtRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.mapping.toModel
+
+@Service
+class CourtsService(private val courtRepository: CourtRepository) {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CourtsService.kt:8]: Courts service or ReferenceDataService?  Dunno just a thought.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/ProbationTeamsService.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.ProbationTeamRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.mapping.toModel
+
+@Service
+class ProbationTeamsService(
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/ProbationTeamsService.kt:8]: Could go into ReferenceDataService?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CourtsServiceTest.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  fun `Should return a list of enabled courts`() {
+    val listOfEnabledCourts = listOf(
+      generateEntity(1L, "COURT1", "One"),
+      generateEntity(2L, "COURT2", "Two"),
+      generateEntity(3L, "COURT3", "Three"),
+    )
+
+    whenever(courtRepository.findAllByEnabledIsTrue()).thenReturn(listOfEnabledCourts)
+
+    Assertions.assertThat(service.getEnabledCourts()).isEqualTo(
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CourtsServiceTest.kt:35]: Minor comment.  `Assertions`. and `Mockito`. could be imported.

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CourtsServiceTest.kt (hunk 3 above)
Comment (timharrison-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CourtsServiceTest.kt:35]: Missed that in this file! I'll sort, as it is in others. IJ must have added.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CourtsService.kt (hunk 1 above)
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CourtsService.kt:8]: I did wonder about combining them, but then thought there is likely to be more court or probation team specific endpoints to come, so separated.

File: src/main/resources/migrations/common/V2024.04.23__baseline_data.sql [hunk 4]
```diff
@@ -39,18 +39,20 @@ values ( 1, 'COURT_HEARING_TYPE', 'APPEAL', 'Appeal', 'TIM', current_timestamp),
...
+       (43, 'NOTIFY_REASON', 'NOTIFY_3RD_PARTY', 'Notify third party', 'TIM', current_timestamp),
+       (44, 'NOTIFY_REASON', 'NOTIFY_COURT', 'Notify court contact', 'TIM', current_timestamp),
+       (45, 'NOTIFY_REASON', 'NOTIFY_PROBATION', 'Notify probation contact', 'TIM', current_timestamp),
+       (46, 'NOTIFY_REASON', 'NOTIFY_PRISON', 'Notify prison contact', 'TIM', current_timestamp),
+       (47, 'NOTIFY_REASON', 'NOTIFY_OWNER', 'Notify owner', 'TIM', current_timestamp),
+       (48, 'HISTORY_TYPE', 'HIST_CREATED', 'Created', 'TIM', current_timestamp),
+       (49, 'HISTORY_TYPE', 'HIST_AMENDED', 'Amended', 'TIM', current_timestamp),
+       (50, 'HISTORY_TYPE', 'HIST_CANCELLED', 'Cancelled', 'TIM', current_timestamp);
 
 alter sequence if exists reference_code_reference_code_id_seq restart with 49;
```
[the remark this answers is not in the document]
Comment (matt-crowson-moj) [reply src/main/resources/migrations/common/V2024.04.23__baseline_data.sql:57]: Good spot.

[the remark this answers is not in the document]
Comment (timharrison-moj) [reply src/main/resources/migrations/common/V2024.04.23__baseline_data.sql]: Done now.  Matt, I'll have a think about IDs - for the next one.

[the remark this answers is not in the document]
Comment (matt-crowson-moj) [reply src/main/resources/migrations/common/V2024.04.23__baseline_data.sql]: But this is because we are not using the sequence Steve, we are bypassing it.

**Review by matt-crowson-moj — APPROVED** [review]
LGTM (the mockito import still needs sort but really very minor)

**Pull request merged**